pkg/controller/nebulascheduledbackup/nebula_scheduled_backup_manager.go

```diff
@@ -0,0 +1,396 @@
+
+	// SyncNebulaScheduledBackup implements the logic for syncing a NebulaScheduledBackup.
+	SyncNebulaScheduledBackup(backup *v1alpha1.NebulaScheduledBackup, successfulBackups, failedBackups, runningBackups []v1alpha1.NebulaBackup, now *metav1.Time) (*kube.ScheduledBackupUpdateStatus, *metav1.Time, error)
+}
+
+var _ Manager = (*scheduledBackupManager)(nil)
+
+type scheduledBackupManager struct {
+	clientSet kube.ClientSet
+}
+
+func NewBackupManager(clientSet kube.ClientSet) Manager {
+	return &scheduledBackupManager{clientSet: clientSet}
+}
+
+func (bm *scheduledBackupManager) CleanupFinishedNBs(scheduledBackup *v1alpha1.NebulaScheduledBackup, successfulJobs, failedJobs []v1alpha1.NebulaBackup) error {
+	klog.Info("Cleaning up previous successful and failed backup jobs")
+	if (scheduledBackup.Spec.MaxSuccessfulNebulaBackupJobs == nil || *scheduledBackup.Spec.MaxSuccessfulNebulaBackupJobs < 0) && (scheduledBackup.Spec.MaxFailedNebulaBackupJobs == nil || *scheduledBackup.Spec.MaxFailedNebulaBackupJobs < 0) {
+		klog.Infof("No limit set for either successful backup jobs or failed backup jobs for nebula scheduled backup %s/%s. Nothing to cleanup.", scheduledBackup.Namespace, scheduledBackup.Name)
+		return nil
+	}
+
+	var jobsToDelete int32
+	if scheduledBackup.Spec.MaxSuccessfulNebulaBackupJobs != nil && *scheduledBackup.Spec.MaxSuccessfulNebulaBackupJobs >= 0 {
+		jobsToDelete = int32(len(successfulJobs)) - *scheduledBackup.Spec.MaxSuccessfulNebulaBackupJobs
+		if jobsToDelete > 0 {
+			err := bm.cleanupNBs(scheduledBackup, successfulJobs, jobsToDelete)
+			if err != nil {
+				return fmt.Errorf("cleanup successful backup jobs failed for nebula scheduled backup %s/%s. err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+			}
+		} else {
+			klog.Infof("Number of successful backup jobs is less then %v. Nothing to cleanup", *scheduledBackup.Spec.MaxSuccessfulNebulaBackupJobs)
+		}
+	} else {
+		klog.Infof("No limit set for successful backup jobs for nebula scheduled backup %s/%s. Nothing to cleanup.", scheduledBackup.Namespace, scheduledBackup.Name)
+	}
+
+	if scheduledBackup.Spec.MaxFailedNebulaBackupJobs != nil && *scheduledBackup.Spec.MaxFailedNebulaBackupJobs >= 0 {
+		jobsToDelete = int32(len(failedJobs)) - *scheduledBackup.Spec.MaxFailedNebulaBackupJobs
+		if jobsToDelete > 0 {
+			err := bm.cleanupNBs(scheduledBackup, failedJobs, jobsToDelete)
+			if err != nil {
+				return fmt.Errorf("cleanup failed backup jobs failed for nebula scheduled backup %s/%s. err: %w", scheduledBackup.Namespace, scheduledBackup.Name, err)
+			}
+		} else {
+			klog.Infof("Number of failed backup jobs is less then %v. Nothing to cleanup", *scheduledBackup.Spec.MaxFailedNebulaBackupJobs)
+		}
+	} else {
+		klog.Infof("No limit set for failed backup jobs for nebula scheduled backup %s/%s. Nothing to cleanup.", scheduledBackup.Namespace, scheduledBackup.Name)
+	}
```

Comment:
remove the else branch to make it concise

Reply:
Removed